### PR TITLE
Migration to Webpack 5

### LINKED ISSUE #5815: Review building an editor guide after upgrading to Webpack 5
## 📝 Provide a description of requested docs changes

Terser is a default minified in Webpack 5. In our docs, we mentioned about optimizing the bundle using something similar to `uglify-js`.

After switching to Webpack 5, we can clean up our guides and recommend using Terser which is available by default.

### PR DESCRIPTION
**Pull request by pomek**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Docs: Upgraded dependencies in the projects. Now we use webpack 5. Closes #5815.

---

### Additional information

Requires: https://github.com/ckeditor/ckeditor5-dev/pull/673.

### CODE REVIEW

**Comment by pomek** [comment]
I reviewed guides and the editor will be built properly.

As for now, we're waiting for the official release. After that, I'll remove `alpha/beta/rc` versions from all files and we will see how the things look.

**Comment by pomek** [comment]
Ready for review.

I stopped the CI since it will fail anyway (requires changes from the dev repository).

File: docs/builds/guides/frameworks/react.md [hunk 1]
```diff
@@ -216,6 +216,8 @@ const { styles } = require( '@ckeditor/ckeditor5-dev-utils' );
 
 Then, add two new elements to the exported object under the `module.rules` array (as new items of the `oneOf` array). These are SVG and CSS loaders required to handle the CKEditor 5 source:
 
+If you use the `postcss-loader` package in version 4+, use the snippet below:
```
Comment (Reinmar) [inline docs/builds/guides/frameworks/react.md:219]: Does postcss-loader come with CRA? I can't see it mentioned anywhere in this guide.

Comment (Reinmar) [reply docs/builds/guides/frameworks/react.md:219]: OK, I can see it comes with CRA. However, currently in version 3.

File: docs/builds/guides/frameworks/react.md [hunk 2]
```diff
@@ -481,14 +519,17 @@ Then add two new elements to the exported object under the `module.rules` array
 				}
 			}
 		},
+		'css-loader',
```
Comment (Reinmar) [inline docs/builds/guides/frameworks/react.md:522]: Why did you change this? CRA1 comes with an old postcss-version so its very unlikely that anyone using it will be also using the latest version of postcss-loader.

Comment (Reinmar) [reply docs/builds/guides/frameworks/react.md:522]: If anything, I'd link here to the CRA2+ section saying:

```
<info-box>
	If you use `postcss-loader` in version 4+ see LINK
</info-box>
```

File: docs/builds/guides/frameworks/vuejs.md [hunk 3]
```diff
@@ -261,12 +261,14 @@ module.exports = {
 			.use( 'postcss-loader' )
 			.loader( 'postcss-loader' )
 			.tap( () => {
-				return styles.getPostCssConfig( {
-					themeImporter: {
-						themePath: require.resolve( '@ckeditor/ckeditor5-theme-lark' ),
-					},
-					minify: true
-				} );
+				return {
```
Comment (Reinmar) [inline docs/builds/guides/frameworks/vuejs.md:264]: Did we have this change tested manually?

File: docs/builds/guides/integration/advanced-setup.md [hunk 4]
```diff
@@ -148,22 +149,30 @@ module.exports = {
...
+                        }
+					}
 				]
 			}
 		]
 	}
 };
 ```
 
+<info-box>
```
Comment (Reinmar) [inline docs/builds/guides/integration/advanced-setup.md:171]: We need to work on the wording here.

But first, I'd like to ensure that:

*   If I change `webpack@5` to `webpack@4` in the `npm i` step **and nothing else**, everything will work fine when I follow this guide? That's how I understand this comment and that'd be great, however...
*   Do we then need the branching for `postcss-loader@4` vs `postcss-loader@3` change in the CRA guide?

File: docs/framework/guides/tutorials/implementing-a-block-widget.md [hunk 5]
```diff
@@ -29,11 +29,11 @@ First, install packages needed to build and set up a basic CKEditor 5 instance.
 
 ```bash
 npm install --save \
-	postcss-loader@3 \
-	raw-loader@3 \
+	postcss-loader@4 \
```
Comment (Reinmar) [inline docs/framework/guides/tutorials/implementing-a-block-widget.md:32]: Missing `css-loader`?

File: package.json [hunk 6]
```diff
@@ -91,21 +92,21 @@
     "@ckeditor/ckeditor5-track-changes": "^23.0.0",
     "@webspellchecker/wproofreader-ckeditor5": "^1.0.5",
     "@wiris/mathtype-ckeditor5": "^7.24.0",
-    "babel-standalone": "^6.26.0",
     "coveralls": "^3.1.0",
-    "css-loader": "^3.5.3",
+    "css-loader": "4.3.0",
```
Comment (Reinmar) [inline package.json:96]: Missing `^`?

**Comment by Reinmar** [comment]
The change looks rathter ok, although, we need to take into account the fact that we can blow up people's setup completely blocking the adoption of CKEditor 5.

What's more, we need to change just too many guides. Integrations, tutorials, quick start, advanced setup – that's a lot to review **and test**. I think we could ask @pkwasnik and @psmyrek together with @ckeditor/qa-team for each person to check one guide/case.

Scenarios to test:

*   I have a fresh new vanilla JS project (so I can use webpack 5) and want to integrate CKEditor 5 via the "from source" scenairo: https://ckeditor.com/docs/ckeditor5/latest/builds/guides/integration/advanced-setup.html#scenario-2-building-from-source
*   Like above but https://ckeditor.com/docs/ckeditor5/latest/framework/guides/quick-start.html + https://ckeditor.com/docs/ckeditor5/latest/framework/guides/tutorials/implementing-a-block-widget.html
*   I have an older webpack 4 based vanilla JS project and I want to use the "from source" method (first scenario) but I cannot upgrade webpack to 5.
*   I have a fresh CRA3 app and I want to follow https://ckeditor.com/docs/ckeditor5/latest/builds/guides/integration/frameworks/react.html.
*   I have an older CRA2 app and I want to follow https://ckeditor.com/docs/ckeditor5/latest/builds/guides/integration/frameworks/react.html.
*   I have a fresh CRA3 app with ejected CRA and all the deps (webpack, loaders, etc.) upgraded and I want to follow the same guide. Here, the question is, can you even upgrade these deps in that environment? CRA3 comes with e.g. postcss-loader@3 and there's postcss-loader@4 on npm. I'm asking because I expect there to be problems while upgrading and perhaps it does not even make sense to mention non-default deps setup in that guide and KISS.

All these scenarios need to be checked, unfortunately. @Mgsy, could you coordinate the testing effort here?

**Comment by Reinmar** [comment]
> All these scenarios need to be checked, unfortunately. @Mgsy, could you coordinate the testing effort here? You will need to let Pawełs know how to use ckeditor5-dev from GH.

We've just talked with @pomek that it'd be good to combine his changes with my changes in #6770 and possibly a change in framework's quick start and tutorials (we should be reusing quick start in tutorials to avoid duplication) and test all these changes at once.

So, I'm calling off the testing :D

**Comment by pomek** [comment]
Good news! We've just released a new version of the [`@ckeditor/ckeditor5-dev-webpack-plugin`](https://www.npmjs.com/package/@ckeditor/ckeditor5-dev-webpack-plugin) package that supports both webpack versions – 4 and 5.

**Comment by pomek** [comment]
Closing due to https://github.com/ckeditor/ckeditor5-dev/pull/673#issuecomment-786715895.
